dfusion_rust_core/src/models/account_state.rs

```diff
@@ -73,7 +73,8 @@ impl AccountState {
         let buy_amounts = results.executed_buy_amounts;
         let sell_amounts = results.executed_sell_amounts;
 
-        for (i, order) in orders.iter().enumerate() {
+        for i in 0..buy_amounts.len() {
+            let order = &orders[i];
```

Comment:
It's not equivalent? If so, I would favour the previous approach

Reply:
Actually it is not equivalent. The number of orders is potentially greater than the number of executed amounts. A perfect example of this is exactly the purpose of the PR (The Trivial Solution) where `orders.len() == 2` and `results.executed_buy_amounts.len() == 0`.